crates/nu-command/src/network/http/client.rs

```diff
@@ -639,3 +646,9 @@ pub fn request_handle_response_headers(
         },
     }
 }
+
+fn retrieve_http_proxy_from_env() -> Option<String> {
+    std::env::vars()
+        .find(|(key, _)| key == "HTTP_PROXY")
```

Comment:
Hi, thanks for your pr, I think it's good to check `ALL_PROXY` too.
In the case if `HTTP_PROXY` exists, use it, or else check `ALL_PROXY`

Reply:
What about `HTTPS_PROXY`? After doing some research on the matter, it seems like tools like curl support HTTP proxy in the lowercase variant (`http_proxy`). Should we look for both variables?

Regarding `ALL_PROXY`, don't we risk an error if using say a FTP proxy to make HTTP calls?

Reply:
@WindSoilder I put `HTTP_PROXY` and `ALL_PROXY` as fallbacks to `http_proxy`.

Reply:
Sorry for late reply.
I think your changes is fine.  Regards to your question:
> Regarding ALL_PROXY, don't we risk an error if using say a FTP proxy to make HTTP calls?

Yeah it is, so that's why we have `http_proxy`. 
And I don't think we need to adapt `HTTPS_PROXY` for now